CRM/Utils/System/WordPress.php

```diff
@@ -789,7 +789,12 @@ public function createUser(&$params, $mail) {
     $creds['user_login'] = $params['cms_name'];
     $creds['user_password'] = $params['cms_pass'];
     $creds['remember'] = TRUE;
-    $user = wp_signon($creds, FALSE);
+
+    // Call wp_signon if we aren't already logged in
+    // For example, we might be creating a new user from the Contact record.
+    if (!current_user_can('create_users')) {
+      wp_signon($creds, FALSE);
```

Comment:
I've always found this problematic but have never opened an issue for it...

It seems like bad practice to log a user in before confirming their email address in the normal WordPress fashion. The default email sent to the new User by `wp_new_user_notification()` below looks like this by default:

```
Username: <username>

To set your password, visit the following address:

https://domain.xyz/wp-login.php?action=rp&key=<key>&login=<username>

https://domain.xyz/wp-login.php
```

I would be in favour of removing it completely.

Reply:
I agree.  I am not seeing the need for this.  The comment mentions that we are not logged in, but to get to the create user account action I cannot think of an instance where we would not be logged in.

Reply:
@kcristiano As discussed, `createUser()` is called from both CiviCRM Admin and front-end Profile Forms. In my opinion, it would be nice if Profile Forms had an option to enable/disable auto-login.